browse/static/js/catalyzex.js

```diff
@@ -19,22 +19,26 @@
   }
 
   const fetchCatalyzeXCode = async () => {
-    const cxApiUrl = `https://www.catalyzex.com/api/code?src=arxiv&paper_arxiv_id=${arxivId}`;
+    const cxApiUrl = new URL("https://www.catalyzex.com/api/code");
+    cxApiUrl.searchParams.set('src', 'arxiv');
+    cxApiUrl.searchParams.set('paper_arxiv_id', arxivId);
```

Comment:
@riltonfranzonee shouldn't we be adding a undefined check here as well, so we are not setting `arxivId` as undefined ?

Reply:
hmm, we're running this on an Arxiv paper page, so I'd say we can safely start with the premise that the `arxivId` is available :)

Reply:
@riltonfranzonee i agree but isn't arxiv_id getting extracted from the DOM [here](https://github.com/gragtah/arxiv-browse/pull/3/files#diff-a555a906314d617df35efac5fc9214dca736ca3f26b3b9c654f671e3e78aeda1R2), what if the DOM gets updated or something like that ?

Reply:
fair enough :)

I've actually updated the PR slightly to retrieve the Arxiv ID from the URL (some other integrations like PWC are doing the same) instead of the DOM and added a validation to return in case no ID is found.